ddp-workspace/projects/ddp-dsm-ui/src/app/clinical-page/clinical-page.component.ts

```diff
@@ -0,0 +1,91 @@
+import {Component, OnInit} from '@angular/core';
+import {Auth} from '../services/auth.service';
+import {ComponentService} from '../services/component.service';
+import {DSMService} from '../services/dsm.service';
+import {RoleService} from '../services/role.service';
+import {Statics} from '../utils/statics';
+import {Utils} from '../utils/utils';
+import {ClinicalOrder} from './clinical-order.model';
+
+@Component( {
+  selector: 'app-clinical-page',
+  templateUrl: './clinical-page.component.html',
+  styleUrls: [ './clinical-page.component.scss' ]
+} )
+export class ClinicalPageComponent implements OnInit {
+  clinicalOrdersArray: ClinicalOrder[];
+  loading: boolean;
+
+  constructor( private dsmService: DSMService, private auth: Auth, private role: RoleService ) {
+  }
+
+  ngOnInit(): void {
+    this.getMercuryOrders();
+  }
+
+  getMercuryOrders(): void {
+    this.loading = true;
+    const realm = localStorage.getItem( ComponentService.MENU_SELECTED_REALM );
+    this.dsmService.getMercuryOrders( realm ).subscribe( {
+      next: data => {
+        const jsonData = data;
+        this.clinicalOrdersArray = [];
+        jsonData.forEach( ( json ) => {
+            const order = ClinicalOrder.parse( json );
+            this.clinicalOrdersArray.push( order );
+          }
+        );
+        this.loading = false;
+      },
+      error: () => {
+
+      }
+    } );
+  }
+
+  getDateFormatted( date: any ): string {
+    return Utils.getLongDateFormatted( date );
+  }
+
+  getOrderStatus( statusDetail: string ): string {
+    if (!statusDetail) {
+      return 'Sent';
+    }
+    return statusDetail;
+
+  }
+
+  public downloadList(): void {
+    const map: { shortId: string; sampleType: string; sample: string; orderDate: string; status: string; orderId: string }[] = [];
+    console.log( this.clinicalOrdersArray );
+    for (const order of this.clinicalOrdersArray) {
+      const dateCreated = this.getDateFormatted( order.orderDate );
```

Comment:
#Optional:
You could also move this for loop's code block into a separate function and then call like this:
for (const order of this.clinicalOrdersArray) {
  doSomething(order)
}